### DISCUSSION TITLE
GNAT in arm64 gcc

### DISCUSSION THREAD

**Discussion by controversial** [Homebrew/discussions#984] [Everyday usage] [answered]
I'm trying to get the GNAT toolchain for compiling Ada (a part of `gcc`) running on my M1 mac but I’ve had a lot of trouble. The Homebrew `gcc` formula currently only includes the C and Fortran compilers, as far as I can tell. Is it at all feasible to ship GNAT as part of the formula? I’ve tried to build it myself in a few different ways but it doesn’t look like it’s possible to build GNAT without already having an older version of GNAT.

cc @fxcoudert

**Comment by carlocab** [comment, marked as answer]
The [formula](https://github.com/Homebrew/homebrew-core/blob/9267d4d07d6f50f4c470237ff518900e69c7ab40/Formula/gcc.rb#L74-L75) currently says
```
    # We avoiding building:
    #  - Ada, which requires a pre-existing GCC Ada compiler to bootstrap
```
If you work out the bootstrap problem, then you can open a pull request to make the changes to the `gcc` formula. In case your PR is not accepted, you can always host your own version of the `gcc` formula in your own tap.
